model/patch/patch.go

```diff
@@ -301,6 +302,18 @@ func (p *Patch) SetParameters(parameters []Parameter) error {
 	)
 }
 
+func (p *Patch) SetDownstreamParameters(parameters []Parameter) error {
```

Comment:
I envision a different use-case: each time we'd call this function it would be with a different set of parameters we'd want to append to the list of parameters. 
In other words, each time a task calls this we'd want to append to the list instead of replacing it.

Reply:
thanks, I'll change it!